### DISCUSSION TITLE
How would one go about discovering possible tools available in defaults?

### DISCUSSION THREAD

**Discussion by slifty** [yannbertrand/macos-defaults#200] [Q&A] [answered]
Thank you for this resource!

I'm curious what *isn't* in here.

Are there "hidden" preference options or is it just a matter of running `defaults read X` and looking at the values that exist / playing around?

**Comment by yannbertrand** [comment]
Hey @slifty thanks for your interest on the project 😄 

So actually I found out it is possible to read all currently set defaults by running `defaults read` with no params. So when I find a Mac setting I find useful I toggle the setting value and run

```shell
defaults read > tmp-defaults-save-value-on.plist
```

I then just need to compare with another `tmp-defaults-save-value-off.plist` file and find the related domain and key I'm looking for!

**Reply by slifty** [reply]
Absolutely - it's great (and I've been having a fun dotfiles adventure so I may contribute a few items to the project once I get my ducks in a row).

That's a really great suggestion thank you!

I'll add that I've learned about a few things that seem to be truly hidden settings (for instance, I don't think there is a UX interface to disable icons on the desktop, but it's possible to do via plist config!)

It sounds like finding *those* settings are just going to be a matter of relying on people who have dug even deeper somehow.

**Reply by yannbertrand** [reply, marked as answer]
Btw we now have a bash script to detect changes when switching an option from UI, take a look at #220 😄
